src/controllers/user.js

```diff
@@ -68,22 +68,21 @@ export const getAll = async (req, res) => {
 }
 
 export const updateById = async (req, res) => {
-  const paramsId = Number(req.params.id)
-  const { id } = req.user
-  const foundUser = await User.findById(paramsId)
+  const { firstName, lastName } = req.body
+  try {
+    validation.update(firstName, lastName)
+    const { id } = req.user
+    const canPatch = validation.validateCanModify(req)
```

Comment:
This function is not properly validating the current user role. Make sure you actually pull the validation data from the authenticated user, not simply the request (which can be forged).